File: examples/app-yaml-config/sdk/mjs/01-load-and-merge.mjs
#!/usr/bin/env node
// F02 / story 01 — load 9 fixtures from server/config/, prove merge + idempotent init.
import { AppYamlConfig } from "@ployglot/app-yaml-config";
import { loadFromConfigDir } from "@ployglot/app-yaml-loader";
import { CONFIG_DIR, resetAndInit, stable } from "./_init.mjs";

const inst = await resetAndInit();
const loaded = await loadFromConfigDir({ configDir: CONFIG_DIR });
const inst2 = await AppYamlConfig.initialize({ loaded });

console.log(stable({
  config_dir: CONFIG_DIR,
  top_level_keys: Object.keys(inst.getAll()).sort(),
  file_count: inst.getOriginalAll().size,
  same_instance: inst === inst2,
}));

File: examples/app-yaml-config/sdk/py/01-load-and-merge.py
#!/usr/bin/env python3
"""F02 / story 01 — load 9 fixtures from server/config/, prove merge + idempotent init."""
from __future__ import annotations

from app_yaml_config import AppYamlConfig
from app_yaml_loader import load_from_config_dir
from _init import CONFIG_DIR, reset_and_init, run, stable


async def main():
    inst = await reset_and_init()
    loaded = load_from_config_dir(config_dir=str(CONFIG_DIR))
    inst2 = await AppYamlConfig.initialize(loaded=loaded)

    print(stable({
        "config_dir": str(CONFIG_DIR),
        "top_level_keys": sorted(inst.get_all().keys()),
        "file_count": len(inst.get_original_all()),
        "same_instance": inst is inst2,
    }))


if __name__ == "__main__":
    run(main)
